### DISCUSSION TITLE
Amazon Print Replica DRM removal?

### DISCUSSION THREAD

**Discussion by tomsem** [noDRM/DeDRM_tools#206] [Q&A]
I believe it was possible to remove DRM from Amazon's 'print replica' (PDF) in the past, with Apprentice Alf's DeDRM. Is this supported with these tools as well? If so, where can I find more information about how to set this up (I normally download from Content and Devices, which yields 'AZW4' file tied to my Fire tablet - but I haven't configured plugin for it, and don't know if that even works).

If not, where can I find information about setting it up with the previous version of the tools? I have looked in the documentation in its GitHub repo but wasn't able to find anything there. IIRC it required Python 2. I'd be willing to take a shot at creating PR for this repo if it is missing and someone can point me to the relevant code. Thanks.

**Comment by ElleKayEm** [comment]
AZW4 is a pdf in a Amazon wrapper.  Use DeDRM as usual and then extract the pdf with the Kindle Unpack plugin.  Won't work with a book tied to a Fire though.  You'll need an E-ink Kindle or Kindle for PC/Mac.  See this post for details on getting books you can decrypt: https://www.mobileread.com/forums/showthread.php?t=283371

**Reply by tomsem** [reply]
Thank you.

E-Ink kindles do not support Print Replica (maybe Kindle Scribe will?), so having one (or several in my case) is of no help in decrypting. 

As for Kindle for Mac/PC, I was hoping to avoid dealing with it and the need to have an old version of it running somewhere, or having to modify default behavior of current to prevent KFX. 

With the current version of Kindle for Mac, I downloaded the Print Replica book I'm trying to decrypt, but importing the resultant '.azw' file results in a un-decrypted '.kfx-zip' format book, which is of no more use than the AZW4 file I download from my Amazon Content page.

I'll follow up on the mobileread thread to see if there is someone who has actually done this successfully.

**Reply by ElleKayEm** [reply]
In that case, sounds like using Kindle for PC/Mac and avoiding kfx is your only option.  I have successfully dealt with AZW4 downloaded with Kindle for PC in the past using DeDRM and the Kindle Unpack plugin.  If you do ask questions at MobileRead, keep in mind that detailed discussion of DRM removal is not allowed there.

**Reply by tomsem** [reply]
Wonder what is involved in decrypting AZW4 (or AZW3/AZW etc) tied to a particular Fire tablet; in principal it does not seem that different than decrypting AZW3 tied to a particular Kindle. I have to think someone has looked at this? 

Relying on some old version of Kindle for Windows/macOS (the newer ones apparently cannot be coerced into suppressing KFX) to keep working seems very fragile. Amazon could pull the plug on old versions at any time.

**Reply by ElleKayEm** [reply]
Getting the key is the issue.  With E-ink Kindles, the key can be derived from the serial number.  With the Kindle for PC/Mac app, DeDRM finds and decrypts the key file.

**Reply by j-howell** [reply]
> Wonder what is involved in decrypting AZW4 (or AZW3/AZW etc) tied to a particular Fire tablet; in principal it does not seem that different than decrypting AZW3 tied to a particular Kindle. I have to think someone has looked at this?

Fire tablets use a variation of the Kindle for Android app. There was a flaw in Kindle for Android that allowed the key to be recovered from it, but Amazon fixed that years ago.

A Fire tablet would need to be rooted for there to be any hope of obtaining the key these days. I am not sure if that is possible but even if so doing that is likely to be beyond the ability of most Fire tablet users.
 
> Relying on some old version of Kindle for Windows/macOS (the newer ones apparently cannot be coerced into suppressing KFX) to keep working seems very fragile. Amazon could pull the plug on old versions at any time.

There are a lot of things Amazon could do to make DRM removal much more difficult.  Enjoy it while you can.

**Reply by tomsem** [reply]
Thanks for confirming all of this. I probably have only about 4 in this format in my Kindle library, not sure it is worth the trouble to set up a decrypting environment just for them. (Note: rooting Fire is possible and straightforward, but someone would still have to demonstrate it helps with decryption, and even then Kindle for Windows/macOS would be preferred by most users).

An alternate approach in this case would be to automate the Kindle app (Keyboard Maestro? Sikuli+Selenium Webdriver?), take screen captures of each page, and make a PDF out of those images. I'm a retired software developer in test, and I think it will be fun to pursue such a project.

**Reply by ElleKayEm** [reply]
Yes, even on a rooted Fire, you'd still need to figure out extracting and presumably decrypting the key.

There was something like your alternative approach for the old comixology web reader.

**Reply by tomsem** [reply]
Unfortunately Kindle web reader does not currently support Print Replica. There are lots of good (cross platform) tools for browser automation. But it looks like _Keyboard Maestro_ will do the job end-to-end, and I will probably find other useful things to do with it.
